etc/init.d/contextBroker.centos

```diff
@@ -94,11 +96,56 @@ broker_start()
         # logs are always managed by CB and written to file inside directory BROKER_LOG_DIR (see /etc/sysconfig/contextBroker) and rotated by logrotate
         BROKER_STDERR_LOG=${BROKER_LOG_DIR}/${NAME}-error.log
         su $BROKER_USER -c "${EXECUTABLE} ${BROKER_OPTS} 1>/dev/null 2>${BROKER_STDERR_LOG}"
+        sleep 2 # wait some time to leave contextbroker start
+        PID=$(cat ${BROKER_PID_FILE})
+        if [[ -z "${PID}" ]]; then
+            printf "%s" "pidfile not found"
+            printf "%s\n" "$(failure)"
+            exit 1
+        else
+            printf "%s\n" "$(success)"
+        fi
+    else
+        printf "%s\n" "Fail - missing ${EXECUTABLE} executable"
+        exit 1
+    fi
+
+}
+
+rotate_cores ()
+{
+    cd /var/cb_cores
+    STR_DATE=$(date +"%Y%m%d_%H%M%S")
+    if [[ $(ls core* 2>/dev/null | wc -l) -gt 0 ]]; then 
+        cp ${BROKER_LOG_DIR}/contextBroker.log . 
+        tar jcf CB_core_${STR_DATE}.tar.bz2 core* contextBroker.log
+        rm -f core* contextBroker.log
+    fi
+
+    TOTAL=$(ls CB_core_* 2>/dev/null | wc -l)
+    KEEP_CORE=8
+    if [[ ${TOTAL} -gt ${KEEP_CORE} ]]; then
+        rm -f $(ls -tr 2>/dev/null |  head -$((${TOTAL} - ${KEEP_CORE})))
+    fi
+
+}
+
+broker_coredump_start()
+{
+    printf "%-50s" "Starting ${NAME}..."
+    if [[ -x ${EXECUTABLE} ]]; then
+        mkdir -p /var/cb_cores &> /dev/null
+        cd /var/cb_cores
+        rotate_cores &
+        ulimit -c unlimited
+        ulimit -n 10240
+        ${EXECUTABLE} ${BROKER_OPTS} -fg &> /dev/null &
         sleep 2 # wait some time to let the contextbroker start
-        PID=$(cat ${BROKER_PID_FILE} 2> /dev/null)
+        ps -ef | grep ${EXECUTABLE} | grep -v grep | awk '{print $2}' > ${BROKER_PID_FILE}
```

Comment:
This here might give problems ...
If somebody is editing contextBroker.cpp, that line **emacs/vi contextBroker.cpp** would turn up in the 'grep' as well ...
And, overwriting the PID file, do we really want this?  It's the brokers responsibility to put its PID in the pid-file ...

Reply:
Makes sense to me

Reply:
In fact, in regular start (no core dumps) we are using cat. https://github.com/telefonicaid/fiware-orion/pull/3065/files#diff-705e27a30a796ccc4adca2c31d736a71R100

Reply:
After discussing with @vgarciag , cat-based solution seems to be fine but, to be more robust, this issue should be solved before changing this PR: https://github.com/telefonicaid/fiware-orion/issues/3075